src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -2297,6 +2297,26 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                 break;
             }
 
+            case NI_Sve_TrigonometricMultiplyAddCoefficient:
+            {
+                assert(isRMW);
+
+                if (targetReg != op1Reg)
+                {
+                    assert(targetReg != op2Reg);
+
+                    GetEmitter()->emitInsSve_R_R(INS_sve_movprfx, EA_SCALABLE, targetReg, op1Reg);
+                }
+
+                HWIntrinsicImmOpHelper helper(this, intrin.op3, node);
+
+                for (helper.EmitBegin(); !helper.Done(); helper.EmitCaseEnd())
+                {
```

Comment:
do you mind sharing the codegen for constant and out-of-bound constant/variable input?

Reply:
Sure. Here's the constant case:
```
movprfx z11, z8
ftmad   z11.d, z11.d, z10.d, #2
```

And out-of-bounds (the intrinsic is gone):
```
G_M52739_IG01:  ;; offset=0x0000
            stp     fp, lr, [sp, #-0x30]!
            stp     x19, x20, [sp, #0x20]
            mov     fp, sp
            add     x1, sp, #48
            str     x1, [fp, #0x18]     // [V14 PSPSym]
            mov     x19, x0
                                                ;; size=24 bbWeight=1 PerfScore 4.50
G_M52739_IG02:  ;; offset=0x0018
            movz    x0, #0x3D00
            movk    x0, #0x7D49 LSL #16
            movk    x0, #686 LSL #32
            movz    x1, #0x5210
            movk    x1, #0x7D49 LSL #16
            movk    x1, #686 LSL #32
            movz    x2, #0xC018      // code for System.String:Concat(System.String,System.String):System.String
            movk    x2, #0x395D LSL #16
            movk    x2, #0x7FFF LSL #32
            ldr     x2, [x2]
            blr     x2
            movz    x1, #0x7DB0      // code for System.Console:WriteLine(System.String)
            movk    x1, #0x39CC LSL #16
            movk    x1, #0x7FFF LSL #32
            ldr     x1, [x1]
            blr     x1
                                                ;; size=64 bbWeight=1 PerfScore 14.00
G_M52739_IG03:  ;; offset=0x0058
            add     x19, x19, #80
            mov     x20, x19
            ldrsb   wzr, [x20]
            add     x0, x20, #32
            movz    x1, #0xCF00      // code for System.Runtime.InteropServices.GCHandle:AddrOfPinnedObject():long:this
            movk    x1, #0x39BF LSL #16
            movk    x1, #0x7FFF LSL #32
            ldr     x1, [x1]
            blr     x1
            ldr     x1, [x20, #0x18]
            add     x0, x0, x1
            sub     x0, x0, #1
            sub     x1, x1, #1
            bic     x0, x0, x1
            ldrsb   wzr, [x0]
            add     x0, x19, #40
            movz    x1, #0xCF00      // code for System.Runtime.InteropServices.GCHandle:AddrOfPinnedObject():long:this
            movk    x1, #0x39BF LSL #16
            movk    x1, #0x7FFF LSL #32
            ldr     x1, [x1]
            blr     x1
            ldr     x1, [x19, #0x18]
            add     x0, x0, x1
            sub     x0, x0, #1
            sub     x1, x1, #1
            bic     x0, x0, x1
            ldrsb   wzr, [x0]
                                                ;; size=108 bbWeight=1 PerfScore 32.00
G_M52739_IG04:  ;; offset=0x00C4
            ldp     x19, x20, [sp, #0x20]
            ldp     fp, lr, [sp], #0x30
            ret     lr
                                                ;; size=12 bbWeight=1 PerfScore 3.00
G_M52739_IG05:  ;; offset=0x00D0
            stp     fp, lr, [sp, #-0x30]!
            stp     x19, x20, [sp, #0x20]
            add     x3, fp, #48
            str     x3, [sp, #0x18]
                                                ;; size=16 bbWeight=0 PerfScore 0.00
G_M52739_IG06:  ;; offset=0x00E0
            adr     x0, [G_M52739_IG04]
                                                ;; size=4 bbWeight=0 PerfScore 0.00
G_M52739_IG07:  ;; offset=0x00E4
            ldp     x19, x20, [sp, #0x20]
            ldp     fp, lr, [sp], #0x30
            ret     lr
                                                ;; size=12 bbWeight=0 PerfScore 0.00
```

And non-const (from the reflection scenario):
```
G_M9075_IG01:  ;; offset=0x0000
            stp     fp, lr, [sp, #-0x10]!
            mov     fp, sp
                                                ;; size=8 bbWeight=1 PerfScore 1.50
G_M9075_IG02:  ;; offset=0x0008
            uxtb    w0, w0
            cmp     w0, #8
            bhs     G_M9075_IG12
            adr     x1, [G_M9075_IG03]
            add     x1, x1, x0,  LSL #3
            br      x1
                                                ;; size=24 bbWeight=1 PerfScore 4.50
G_M9075_IG03:  ;; offset=0x0020
            ftmad   z0.d, z0.d, z1.d, #0
            b       G_M9075_IG11
                                                ;; size=8 bbWeight=1 PerfScore 4.00
G_M9075_IG04:  ;; offset=0x0028
            ftmad   z0.d, z0.d, z1.d, #1
            b       G_M9075_IG11
                                                ;; size=8 bbWeight=1 PerfScore 4.00
G_M9075_IG05:  ;; offset=0x0030
            ftmad   z0.d, z0.d, z1.d, #2
            b       G_M9075_IG11
                                                ;; size=8 bbWeight=1 PerfScore 4.00
G_M9075_IG06:  ;; offset=0x0038
            ftmad   z0.d, z0.d, z1.d, #3
            b       G_M9075_IG11
                                                ;; size=8 bbWeight=1 PerfScore 4.00
G_M9075_IG07:  ;; offset=0x0040
            ftmad   z0.d, z0.d, z1.d, #4
            b       G_M9075_IG11
                                                ;; size=8 bbWeight=1 PerfScore 4.00
G_M9075_IG08:  ;; offset=0x0048
            ftmad   z0.d, z0.d, z1.d, #5
            b       G_M9075_IG11
                                                ;; size=8 bbWeight=1 PerfScore 4.00
G_M9075_IG09:  ;; offset=0x0050
            ftmad   z0.d, z0.d, z1.d, #6
            b       G_M9075_IG11
                                                ;; size=8 bbWeight=1 PerfScore 4.00
G_M9075_IG10:  ;; offset=0x0058
            ftmad   z0.d, z0.d, z1.d, #7
                                                ;; size=4 bbWeight=1 PerfScore 3.00
G_M9075_IG11:  ;; offset=0x005C
            ldp     fp, lr, [sp], #0x10
            ret     lr
                                                ;; size=8 bbWeight=1 PerfScore 2.00
G_M9075_IG12:  ;; offset=0x0064
            bl      CORINFO_HELP_THROW_ARGUMENTOUTOFRANGEEXCEPTION
            brk_windows #0
                                                ;; size=8 bbWeight=0 PerfScore 0.00
```